autoload/ale/sign.vim

```diff
@@ -67,10 +67,14 @@ endfunction
 " Given a loclist, combine the loclist into a list of signs such that only
 " one sign appears per line. Error lines will take precedence.
 " The loclist will have been previously sorted.
-function! ale#sign#CombineSigns(loclist) abort
+function! ale#sign#CombineSigns(buffer, loclist) abort
     let l:signlist = []
 
     for l:obj in a:loclist
+        if has_key(l:obj, 'bufnr') && l:obj.bufnr >= 0 && l:obj.bufnr != a:buffer
```

Comment:
I don't think errors from other files should be set in the first place. If you want to see problems in those files, go look at those files. Checking a file should only return problems for that one file.